### PR TITLE
fix(std/http2): Release window capacity back to remote stream

### LINKED ISSUE #24552: large multi-chunk GRPC responses get stuck
Version: Deno 1.45.1

If a response has more than `u16::MAX` many bytes, these polls get stuck.

https://github.com/denoland/deno/blob/8cbf81cf0840aa9c2e7c8e845748b422827c9b88/ext/node/ops/http2.rs#L455-L477

I created a simple pingpong project to reproduce this using a python grpc server: https://github.com/rnbguy/deno-grpc-multi-chunk

Using this example, I realized, Deno parses `u16::MAX` many bytes in four chunks - `16_384`, `16_384`, `16_384` and `16_383` - then the poll gets stuck if there are more bytes pending.

Ref: https://github.com/denoland/deno/pull/24147#issuecomment-2225495224

### PR DESCRIPTION
**Pull request by rnbguy**
This PR adds logic to release window capacity after reading the chunks from the stream. Without it, large response (more than `u16::MAX`) may fill up the capacity and the whole response can't be read.

Closes #24552
Closes https://github.com/denoland/deno/issues/24305

### CODE REVIEW

File: ext/node/ops/http2.rs [hunk 1]
```diff
@@ -466,7 +466,9 @@ fn poll_data_or_trailers(
     }
     if let Poll::Ready(data) = body.poll_data(cx) {
       if let Some(data) = data {
-        return Poll::Ready(Ok(DataOrTrailers::Data(data?)));
+        let data = data?;
+        body.flow_control().release_capacity(data.len())?;
+        return Poll::Ready(Ok(DataOrTrailers::Data(data)));
       }
       // If data is None, loop one more time to check for trailers
       continue;
```
Comment (rnbguy) [inline ext/node/ops/http2.rs:474]: I have a question (unrelated to the PR) - why can't we return `Poll::Pending` here and remove the outer loop? I see that this method is only used with `poll_fn`. This way we can keep this method loop free.

**Review by satyarohith — COMMENTED** [review]
Nice work @rnbguy! This also fixes https://github.com/denoland/deno/issues/24305

Can you add a test for this issue? Example https://github.com/denoland/deno/pull/24390/files#diff-d30b0be158290060aef0491e9888c5bd98e1274eb2196684a8c12feaa7f29cf7

File: ext/node/ops/http2.rs (hunk 1 above)
Comment (satyarohith) [reply ext/node/ops/http2.rs:474]: yes, we can remove that.

Comment (rnbguy) [reply ext/node/ops/http2.rs]: 1a33977

**Review by satyarohith — APPROVED** [review]
Nice test!

Thanks for your contribution @rnbguy!

LGTM.

**Pull request merged**